driver-core/src/main/java/com/datastax/driver/core/DataTypeParser.java

```diff
@@ -0,0 +1,255 @@
+            .build();
+
+    static DataType parse(String toParse, Cluster cluster, String currentKeyspaceName, Map<String, UserType> userTypes, boolean frozen) {
+
+        Parser parser = new Parser(toParse, 0);
+        String type = parser.parseTypeName();
+
+        DataType nativeType = NATIVE_TYPES_MAP.get(type.toLowerCase());
+        if (nativeType != null)
+            return nativeType;
+
+        if (type.equalsIgnoreCase(LIST)) {
+            List<String> parameters = parser.parseTypeParameters();
+            if (parameters.size() != 1)
+                throw new DriverInternalError(String.format("Excepting single parameter for list, got %s", parameters));
+            DataType elementType = parse(parameters.get(0), cluster, currentKeyspaceName, userTypes, false);
+            return list(elementType, frozen);
+        }
+
+        if (type.equalsIgnoreCase(SET)) {
+            List<String> parameters = parser.parseTypeParameters();
+            if (parameters.size() != 1)
+                throw new DriverInternalError(String.format("Excepting single parameter for set, got %s", parameters));
+            DataType elementType = parse(parameters.get(0), cluster, currentKeyspaceName, userTypes, false);
+            return set(elementType, frozen);
+        }
+
+        if (type.equalsIgnoreCase(MAP)) {
+            List<String> parameters = parser.parseTypeParameters();
+            if (parameters.size() != 2)
+                throw new DriverInternalError(String.format("Excepting two parameters for map, got %s", parameters));
+            DataType keyType = parse(parameters.get(0), cluster, currentKeyspaceName, userTypes, false);
+            DataType valueType = parse(parameters.get(1), cluster, currentKeyspaceName, userTypes, false);
+            return map(keyType, valueType, frozen);
+        }
+
+        if (type.equalsIgnoreCase(FROZEN)) {
+            List<String> parameters = parser.parseTypeParameters();
+            if (parameters.size() != 1)
+                throw new DriverInternalError(String.format("Excepting single parameter for frozen keyword, got %s", parameters));
+            return parse(parameters.get(0), cluster, currentKeyspaceName, userTypes, true);
+        }
+
+        if (type.equalsIgnoreCase(TUPLE)) {
+            List<String> rawTypes = parser.parseTypeParameters();
+            List<DataType> types = new ArrayList<DataType>(rawTypes.size());
+            for (String rawType : rawTypes) {
+                types.add(parse(rawType, cluster, currentKeyspaceName, userTypes, false));
+            }
+            return cluster.getMetadata().newTupleType(types);
+        }
+
+        // return a custom type for the special empty type
+        // so that it gets detected later on, see TableMetadata
+        if(type.equalsIgnoreCase(EMPTY))
+            return custom(type);
+
+        // Try UDTs
+        if (userTypes != null) {
+            UserType userType = userTypes.get(Metadata.handleId(type));
```

Comment:
It does seem like an inconsistency, but I'm guessing it needs to be done otherwise someone could throw a '>' in a quoted identifier to throw the parsing off.

For example:

```
cqlsh:test> create type "Mixe>dCase>Type"(i int);
cqlsh:test> create table foo3 ("Key" int primary key, v list<frozen<"Mixe>dCase>Type">>);
cqlsh:test> select * from system_schema.columns where keyspace_name='test' and table_name='foo3';

 keyspace_name | table_name | column_name | clustering_order | column_name_bytes | kind          | position | type
---------------+------------+-------------+------------------+-------------------+---------------+----------+---------------------------------
          test |       foo3 |         Key |             none |          0x4b6579 | partition_key |        0 |                             int
          test |       foo3 |           v |             none |              0x76 |       regular |       -1 | list<frozen<"Mixe>dCase>Type">>
```

This typename seems to throw off of the DataTypeParser due the appearance of '>' in the typename:

```
    657    [main] ERROR com.datastax.driver.core.SchemaParser  - Error parsing schema for table test.foo3: Cluster.getMetadata().getKeyspace("test").getTable("foo3") will be missing or incomplete
com.datastax.driver.core.exceptions.DriverInternalError: Excepting single parameter for list, got [frozen<"Mixe>, dCase]
	at com.datastax.driver.core.DataTypeParser.parse(DataTypeParser.java:84)
	at com.datastax.driver.core.TableMetadata.build(TableMetadata.java:188)
	at com.datastax.driver.core.SchemaParser.buildTables(SchemaParser.java:171)
	at com.datastax.driver.core.SchemaParser.buildKeyspaces(SchemaParser.java:123)
	at com.datastax.driver.core.SchemaParser.refresh(SchemaParser.java:64)
	at com.datastax.driver.core.ControlConnection.refreshSchema(ControlConnection.java:329)
	at com.datastax.driver.core.ControlConnection.tryConnect(ControlConnection.java:260)
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:187)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:79)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1398)
	at com.datastax.driver.core.Cluster.init(Cluster.java:167)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:346)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:319)
	at com.datastax.driver.core.Cluster.connect(Cluster.java:257)
```